### PR TITLE
exposed the number of instances of a verticle that were started

### PR DESCRIPTION
**Pull request by jtruelove**
Signed-off-by: Jeremy Truelove <jtruelove@gmail.com>

### CODE REVIEW

**Comment by jtruelove** [comment]
anything else needed on this?

**Comment by vietj** [comment]
Hi,

we will review and validate it for the next milestone4 hopefully.

File: src/main/java/io/vertx/core/impl/ContextImpl.java [hunk 1]
```diff
@@ -279,5 +279,11 @@ private void setTCCL() {
     Thread.currentThread().setContextClassLoader(tccl);
   }
 
-
+  public int getInstanceCount(){
+    if(deployment == null || deployment.deploymentOptions() == null) {
+      // implicitly there is the current instance of the verticle running always
```
Comment (purplefox) [inline src/main/java/io/vertx/core/impl/ContextImpl.java]: If deployment == null then it should return zero as it's possible to have a context without a verticle (e.g. in the embedded case)

File: src/test/java/io/vertx/test/core/DeploymentTest.java [hunk 2]
```diff
@@ -987,6 +987,46 @@ public void testUndeployAllNoDeployments() throws Exception {
...
+    vertx.eventBus().consumer("instanceCount", event -> {
+      messageCount.incrementAndGet();
+      totalReportedInstances.addAndGet((int)event.body());
+      if(messageCount.intValue() == 3) {
+        assertEquals(9, totalReportedInstances.get());
+      }
+    });
+
+    vertx.deployVerticle(TestVerticle3.class.getCanonicalName(), new DeploymentOptions().setInstances(3), ar -> {
+      assertTrue(ar.succeeded());
```
Comment (purplefox) [inline src/test/java/io/vertx/test/core/DeploymentTest.java]: Not sure this test is valid as the deployment handler could be called before the message consumer has got all the instanceCount messages.

Better to put the testComplete() in the message handler.

File: src/main/java/io/vertx/core/impl/ContextImpl.java (hunk 1 above)
Comment (jtruelove) [inline src/main/java/io/vertx/core/impl/ContextImpl.java]: Gotcha, so should the logic then be

if(deployment == null) {
   return 0;
} else if ( deployment.deploymentOptions == null) {
  return 1;
}

File: src/test/java/io/vertx/test/core/DeploymentTest.java (hunk 2 above)
Comment (jtruelove) [inline src/test/java/io/vertx/test/core/DeploymentTest.java]: kk, easy fix

**Comment by jtruelove** [comment]
updated with feedback

**Pull request merged**